locations/Snakefile

```diff
@@ -94,7 +171,58 @@ rule import_state:
                 --scale tract \
                 --identifier-from GEOID \
                 --hierarchy 'country=>united_states,state=>'{state_name:q} \
+                --hierarchy-by-feature {input.hierarchy} \
                 --point-from INTPTLON INTPTLAT \
                 --simplified-polygons {input.simplified:q} \
                 --if-exists update
         """)
+
+rule import_puma:
+    message: "Importing {wildcards.vintage} Census PUMAs into ID3C for {wildcards.state_fips_code}"
+    input:
+        original   = ancient("data/puma/tl_{vintage}_{state_fips_code}_puma10.zip"),
+        simplified = ancient("data/puma/cb_{vintage}_{state_fips_code}_puma10_500k.zip"),
+    output:
+        touch("log/import/{vintage}_{state_fips_code}_puma")
+    run:
+        state_name = hierarchy_state(fips_to_name(wildcards.state_fips_code))
+        shell("""
+            {id3c:q} location import \
+                {input.original:q} \
+                --scale puma \
+                --identifier-from GEOID10 \
+                --hierarchy 'country=>united_states,state=>'{state_name:q} \
+                --simplified-polygons {input.simplified:q} \
+                --if-exists update
+        """)
+
+rule cra_to_neighborhood_district:
+    message: "Combining Community reporting areas(CRAs) features into neighborhood district features"
+    input:
+        geojson = ancient(rules.download_community_reporting_areas.output)
+    output:
+        geojson = "data/neighborhood_district/seattle_neighborhood_district.geojson"
+    shell:
+        """
+        ogr2ogr \
+            -f GeoJSON \
+            {output.geojson} {input.geojson} \
+            -dialect sqlite \
+            -sql "SELECT replace(replace(NEIGHDIST,' ','_'), '/', '_') as NEIGHDIST, ST_Union(geometry) as geometry FROM seattle_cra GROUP BY CRA_GRP"
```

Comment:
Hmm, this throws an error for me:

```
ERROR 1: In ExecuteSQL(): sqlite3_prepare_v2(SELECT replace(replace(NEIGHDIST,' ','_'), '/', '_') as NEIGHDIST, ST_Union(geometry) as geometry FROM seattle_cra GROUP BY CRA_GRP):
  no such table: seattle_cra
```

Reply:
I think this might be version problem?
According to the GDAL [docs](https://gdal.org/drivers/vector/geojson.html): 
```
Starting with GDAL 2.2, the layer name is built with the following logic:
1. If a “name” member is found at the FeatureCollection level, it is used.
2. Otherwise if the filename is regular (ie not a URL with query parameters), then the filename without extension and path is used as the layer name.
3. Otherwise OGRGeoJSON is used.
```

Reply:
Hmm:

```
tom@whunk locations (pumas=) $ apt policy gdal-bin
gdal-bin:
  Installed: 2.2.3+dfsg-2
  …
```

Reply:
Running `ogrinfo -ro data/cra/seattle_cra.geojson` gives this layer name for me:
```
INFO: Open of `data/cra/seattle_cra.geojson'
      using driver `ESRIJSON' successful.
1: seattle_cra (Polygon)
```

Reply:
Looks like the table name that works for me is `OGRGeoJSON`.

Reply:
Ah, which makes sense given the docs you quoted, since there is no `type: FeatureCollection` field in the input GeoJSON (making it not a valid GeoJSON!).

Reply:
```
$ ogrinfo --version
GDAL 2.2.3, released 2017/11/20
$ ogrinfo -ro data/cra/seattle_cra.geojson
INFO: Open of `data/cra/seattle_cra.geojson'
      using driver `GeoJSON' successful.
1: OGRGeoJSON (Polygon)
```

Reply:
It looks like changing the download URL parameter from `f=json` to `f=geojson` gets us standard GeoJSON instead of ESRI's JSON format.

Reply:
```
$ ogrinfo --version
GDAL 2.4.2, released 2019/06/28
```

Reply:
If I download with `f=geojson`, then `ogrinfo` returns `seattle_cra` for me too.